pkg/api/cluster_types_test.go

```diff
@@ -84,7 +85,79 @@ func TestGetAvailableStrimziVersions(t *testing.T) {
 	}
 }
 
-func TestGetAvailableAndReadyStrimziVersions(t *testing.T) {
+func Test_Cluster_GetLatestAvailableStrimziVersion(t *testing.T) {
+	tests := []struct {
+		name    string
+		cluster func() *Cluster
+		want    *StrimziVersion
+		wantErr bool
+	}{
+		{
+			name: "When cluster has a non empty list of available strimzi versions the latest one is returned",
+			cluster: func() *Cluster {
+				inputStrimziVersions := []StrimziVersion{
+					{Version: "v3", Ready: true},
+					{Version: "v6", Ready: false},
+					{Version: "v7", Ready: false},
+				}
+				inputStrimziVersionsJSON, err := json.Marshal(inputStrimziVersions)
+				if err != nil {
+					panic(err)
+				}
+				res := Cluster{AvailableStrimziVersions: inputStrimziVersionsJSON}
+				return &res
+			},
+			want:    &StrimziVersion{Version: "v7", Ready: false},
+			wantErr: false,
+		},
+		{
+			name: "When cluster has an empty list of available strimzi nil is returned",
+			cluster: func() *Cluster {
+				inputStrimziVersions := []StrimziVersion{}
+				inputStrimziVersionsJSON, err := json.Marshal(inputStrimziVersions)
+				if err != nil {
+					panic(err)
+				}
+				res := Cluster{AvailableStrimziVersions: inputStrimziVersionsJSON}
+				return &res
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "When cluster has a nil list of available strimzi the empty list is returned",
+			cluster: func() *Cluster {
+				res := Cluster{AvailableStrimziVersions: nil}
+				return &res
+			},
+			want:    nil,
+			wantErr: false,
+		},
+		{
+			name: "When cluster has an invalid JSON an error is returned",
+			cluster: func() *Cluster {
+				res := Cluster{AvailableStrimziVersions: []byte(`"keyone": valueone`)}
+				return &res
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := gomega.NewWithT(t)
```

Comment:
Can we run the test in parallel? If yes, please add `t.Parallel()` here

Reply:
I think it probably can be done. This makes me wonder what's the strateg to follow for this however. It's hard to enforce. It also makes me wodner whether we really need to worry about this until performance becomes a real concern.

Reply:
> until performance becomes a real concern

I usually agree about this when the change is big, but since the change is just one row, I really prefer to avoid waiting to have a problem to add a possible solution.

Moreover, running the tests in parallel has the added value of testing that the tested code works as expected when invoked in parallel too.

Reply:
See https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1162#issuecomment-1173934813

Reply:
> Moreover, running the tests in parallel has the added value of testing that the tested code works as expected when invoked in parallel too.

That is a good point

Reply:
I added `Parallel` calls to the subtests in the newly written tests.
I still wonder about the cost of adding this at cognitive load level vs the real need for parallel tests.